keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlDeliveryConfigRepository.kt

```diff
@@ -568,11 +568,17 @@ class SqlDeliveryConfigRepository(
         .where(
           ENVIRONMENT_ARTIFACT_CONSTRAINT.ENVIRONMENT_UID.eq(environmentUID),
           ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_VERSION.eq(artifactVersion),
-          (ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.eq(artifactReference)
-            // for backwards comparability
-            .or(ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.isNull)),
           ENVIRONMENT_ARTIFACT_CONSTRAINT.TYPE.eq(type)
         )
+        .let {
+          if (artifactReference != null) {
+            it.and(
+              ENVIRONMENT_ARTIFACT_CONSTRAINT.ARTIFACT_REFERENCE.eq(artifactReference)
+            )
+          } else {
+            it
```

Comment:
few things:
1. can you please add a comment here if possible? 
2. will else {it} handle the `null` case?
3. this fix should be added to line 929 as well :)

Reply:
1. Sure.
2. If the artifact reference is `null`, we simply won't include it in the query, so it won't matter what's in the DB (same as previous behavior, i.e. backwards compatible).
3. Thanks, will do!

Reply:
Actually, for number 3, I don't think we can do the same thing because otherwise there's no filtering based on artifact in the query. In fact, I think that argument should not be nullable in that method.